src/mesh/forest/tree.hpp

```diff
@@ -0,0 +1,131 @@
+//========================================================================================
+// (C) (or copyright) 2023. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef MESH_FOREST_TREE_HPP_
+#define MESH_FOREST_TREE_HPP_
+
+#include <array>
+#include <map>
+#include <memory>
+#include <set>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include "basic_types.hpp"
+#include "defs.hpp"
+#include "mesh/forest/cell_center_offsets.hpp"
+#include "mesh/forest/relative_orientation.hpp"
+#include "mesh/logical_location.hpp"
+#include "utils/bit_hacks.hpp"
+#include "utils/indexer.hpp"
+
+namespace parthenon {
+namespace forest {
+
+// We don't explicitly allow for periodic boundaries, since we can encode periodicity
+// through connectivity in the forest
+class Tree : public std::enable_shared_from_this<Tree> {
+  // This allows us to ensure that Trees are only created as shared_ptrs
```

Comment:
Never mind the comment above, the explanation is already here.
When reviewing I should probably read the header files first and then the impl.